src/app/core/entity-list/duplicate-records/duplicate-records.service.ts

```diff
@@ -10,11 +11,17 @@ export class DuplicateRecordService {
   constructor(
     private entitymapperservice: EntityMapperService,
     private entityService: EntitySchemaService,
+    private alertService: AlertService,
   ) {}
 
-  async duplicateRecord(sourceData: Entity[]) {
-    const duplicateData = this.clone(sourceData);
-    return await this.entitymapperservice.saveAll(duplicateData);
+  async duplicateRecord(sourceData: Entity | Entity[]) {
+    const entities = Array.isArray(sourceData) ? sourceData : [sourceData];
+    const duplicateData = this.clone(entities);
+
+    await this.entitymapperservice.saveAll(duplicateData);
+
```

Comment:
Maybe we can navigate to the newly cloned entity? (if duplicateRecord() is called with a special navigateTo parameter that we only set to true for single clone)